src/coreclr/jit/compiler.h

```diff
@@ -8224,6 +8192,29 @@ class Compiler
         return eeGetEEInfo()->targetAbi == abi;
     }
 
+#if defined(FEATURE_EH_X86_FRAMES)
+    bool UsesFunclets()
+    {
+        return IsTargetAbi(CORINFO_NATIVEAOT_ABI);
+    }
+
+    bool UsesCallfinallyThunks()
+    {
+        // Generate call-to-finally code in "thunks" in the enclosing EH region, protected by "cloned finally" clauses.
+        return UsesFunclets();
```

Comment:
This incorporates the change from https://github.com/dotnet/runtime/pull/99718 which was not approved yet.